apps/aecore/src/aec_conductor.erl

```diff
@@ -443,18 +443,21 @@ update_chain_metrics(Block) ->
                            aec_blocks:height(Block)).
 
 
-cleanup_after_worker(Info) ->
+cleanup_after_worker(Pid, Info) ->
     case Info#worker_info.timer of
         no_timer -> ok;
         {t, TRef} -> timer:cancel(TRef)
     end,
     demonitor(Info#worker_info.mon, [flush]),
+    %% Flush messages from this worker.
```

Comment:
The flush is unnecessary when we have received the answer and there is a race for the shutdown in `kill_worker/3`. Better place this flush after the `exit(Pid, shutdown)` instead.